sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningAwareFileIndex.scala

```diff
@@ -221,6 +221,16 @@ abstract class PartitioningAwareFileIndex(
         if (!fs.isDirectory(userDefinedBasePath)) {
           throw new IllegalArgumentException(s"Option '$BASE_PATH_PARAM' must be a directory")
         }
+        def qualifiedPath(path: Path): Path = path.makeQualified(fs.getUri, fs.getWorkingDirectory)
+
+        val qualifiedBasePath = qualifiedPath(userDefinedBasePath)
+        rootPaths.find(p => !qualifiedPath(p).toString.
+          startsWith(qualifiedBasePath.toString)) match {
```

Comment:
The indent is off here. But can you just use `.find(...).foreach(rp => ...`?
Or `require(rootPaths.forall(p => qualifiedPath(p)...), "error message")`